### DISCUSSION TITLE
make RGBA image and not Indexed

### DISCUSSION THREAD

**Discussion by Tosyk** [ImageMagick/ImageMagick#4292] [Digital Image Processing] [answered]
whenever I conver rgba image to PNG I get indexed flattened image according to photoshop:
![image](https://user-images.githubusercontent.com/24209966/135269797-3ae2b159-56f3-4186-a556-e45c8c99c7fc.png)
so there's no layers to edit. I have to convert it to another Mode:
![image](https://user-images.githubusercontent.com/24209966/135269917-0d980090-a24a-4781-9a51-9cbd22d83064.png)
what option to use in magic to get the rgba image instead of indexed?

**Comment by snibgo** [comment, marked as answer]
What version of IM, on what platform? What was your command?

By default, IM will save PNG as indexed if that doesn't lose data. You can use an output prefix to force a number of channels and bit-depth:
```
magick in.png PNG24:out.png
magick in.png PNG32:out.png
magick in.png PNG48:out.png
magick in.png PNG64:out.png
```
"24" is 3 channels (RGB) with 8 bits each.

"32" is 4 channels (RGBA) with 8 bits each.

"48" is 3 channels (RGB) with 16 bits each.

"64" is 4 channels (RGBA) with 16 bits each.

**Reply by Tosyk** [reply]
wow, I didn't know about it. thank you!
